TEXroot relative to project settings

wronex opened the pull request:
I've modified `get_tex_root()` to take the project-file's location into account when searching for the master file for projects that define a `"TEXroot"` setting. This solves two issues. Firstly, it is now possible to compile multi-file projects from included tex files that are located in subdirectories. Secondly, it is now possible to open multiple projects simultaneously (in different windows) and compile them. Issue #152 is related.

msiniscalchi commented:
I would like to merge this, but I don't fully understand how it works. In particular, would the code for reading the "global" settings ever be reached? Are "global" settings the ones from LaTeXTools.sublime-settings or other settings files? If so, I don't see how it would make sense to specify the TeX root in a global settings file...

I'm sorry if the above is naive---I don't actually use projects myself, though I can see they may be very useful to define project-specific things such as build engines (this should be possible right now), or the  location of bib files (not yet implemented).

ig0774 commented:
@msiniscalchi: The "global" settings are the settings specific to the view. These default to the values specified for the project in its `settings` section. Apart from that, they could also be set by invoking the API. They are definitely *not* loaded from any settings file. Basically that block is pretty much dead code unless somebody somewhere was relying on setting `TEXroot` even though they didn't have a project file, e.g., in a custom `EventListener` or something similar.

Incidentally, I don't think build engines are configurable in project files yet, at least not while making use of `makePDF.py`...

wronex commented:
It was merely an attempt at backwards compatibility... but I've updated the pull request and removed the dead code. The code is now ST2 safe, but, the issue still remains for ST2 users since ST2 does not support ``view.window().project_file_name()``.

In hindsight, I think that ``% !TEX root = `` comments are a better solution. Maybe you should ignore this pull request and instead remove support for settings files.

ig0774 commented:
Since I've merged #629 which does the same thing but also works on ST2, I'm closing this.
